### PR TITLE
docs: move Breadcrumb and ToC component docs for i18n efforts

### LINKED ISSUE #20854: (docs) Move content about Breadcrumbs and Table of Contents from "docs-and-website-components"
## Summary

The [Docs and Website Components page](https://www.gatsbyjs.org/contributing/docs-and-website-components/) has a section called "other available components", which lists breadcrumbs and table of contents as components you can import to your page.

But that those components are *already* included by default on most docs pages and don't need to be manually imported to markdown. The relevant information is:

- For `<Breadcrumbs>`, that you can change the name of the component in the sidebar/breadcrumb.
- For `<TableOfContents>` that you can disable it using frontmatter.

I think it would make more sense to remove these sections and put the relevant information in [Docs Contributions](https://www.gatsbyjs.org/contributing/docs-contributions/), creating:

- a "table of contents" or "frontmatter" section with all the available options
- a "creating a page" section with all the steps/options available for creating a new page, including adding it to the sidebar.

### Motivation

1. I think it makes more sense this way
2. I'd rather not import components we don't need (it makes it easier for localization)

### PR DESCRIPTION
**Pull request by gillkyle**
## Description

This moves info about the Breadcrumb and Table of Contents components into a more general page of thee contributing docs which makes more sense since they aren't things you'd likely import into a doc.

It removes the imports for the components to help make i18n efforts easier. I also removed the sample renderings since they don't seem like super useful information on this page and probably won't be encountered by contributors anyway.

## Related Issues

Fixes #20854

### CODE REVIEW

File: docs/contributing/docs-contributions.md [hunk 1]
```diff
@@ -103,6 +103,48 @@ It can be necessary to change a heading within the docs. It's important to note
...
 
+## Configuring site navigation
+
+The docs include custom built components to aid with navigation. In order to customize the navigation experience, these components allow some configurations without changing any of the React code.
+
+### Adjusting breadcrumb titles
+
+The `<Breadcrumb />` component is used in layout files to display the hierarchy of pages a user is currently browsing on at the top of each doc.
+
+To alter the title of a doc that is displayed in the Breadcrumb component, a `breadcrumbTitle` is supported as a key in the [sidebar YAML files](https://github.com/gatsbyjs/gatsby/tree/master/www/src/data/sidebars). It is commonly used to provide an abbreviated version of a doc's title when displayed next to its parent page title, e.g. shortening "Adding a Custom webpack Config" to "webpack Config".
```
Comment (AishaBlake) [inline docs/contributing/docs-contributions.md]:
```suggestion
To alter the title of a doc that is displayed in the Breadcrumb component, `breadcrumbTitle` is supported as a key in the [sidebar YAML files](https://github.com/gatsbyjs/gatsby/tree/master/www/src/data/sidebars). It is commonly used to provide an abbreviated version of a doc's title when displayed next to its parent page title, e.g. shortening "Adding a Custom webpack Config" to "webpack Config".
```

This feels a little awkward to me but it's entirely personal preference.

Comment (gillkyle) [reply docs/contributing/docs-contributions.md]: I trust your judgment and think that sounds great!

**Comment by AishaBlake** [comment]
@gillkyle mind resolving those conflicts? I think this is good to go otherwise!

**Comment by gillkyle** [comment]
@AishaBlake Just resolved!

**Pull request merged**